clang/test/CIR/CodeGen/inlineAttr.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clang_cc1 -O2 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s -check-prefix=CIR
+// RUN: %clang_cc1 -O2 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=LLVM
+
+
+inline int s0(int a, int b) {
+  int x = a + b;
+  return x;
+}
+
+__attribute__((noinline))
+int s1(int a, int b) {
+  return s0(a,b);
+}
+
+__attribute__((always_inline))
+int s2(int a, int b) {
+  return s0(a,b);
+}
+
+int s3(int a, int b) {
+  int x = a + b;
+  return x;
+}
+
+
+// CIR:   cir.func linkonce_odr @_Z2s0ii(%arg0:{{.*}}, %arg1:{{.*}} -> {{.*}} extra( {inline = #cir.inline<hint>} )
+// CIR:   cir.func @_Z2s1ii(%arg0:{{.*}}, %arg1:{{.*}} -> {{.*}} extra( {inline = #cir.inline<no>} )
```

Comment:
Why do we get spaces like this: "( {"? 

Also, why do we get `extra( {inline = #cir.inline<no>} )` instead of `extra({#cir.inline<no>})`? Seems like the extra `inline = ...` is redundant, is that a property of how dictionaries operate with their elements? I don't see other Enums elsewhere in CIR getting the same printing behavior.

Reply:
That's exactly this comment discusses: https://github.com/llvm/clangir/pull/134/files#r1256536825

> Also, why do we get extra( {inline = #cir.inline<no>} ) instead of extra({#cir.inline<no>})

Similarly with the comment above, I would like to reuse the MLIR built-in printer and parser to process a named dictionary attribute. Note that our CIR module attribute is also printed in that way.